src/main/scala/viper/gobra/translator/encodings/slices/SliceEncoding.scala

```diff
@@ -144,6 +146,134 @@ class SliceEncoding(arrayEmb : SharedArrayEmbedding) extends LeafTypeEncoding {
+            _ <- write(vpr.Exhale(vpr.LeCmp(vpr.IntLit(0)(pos, info, errT), vprLength)(pos, info, errT))(pos, info, errT)) // 0 <= len
+            _ <- write(vpr.Exhale(vpr.LeCmp(vpr.IntLit(0)(pos, info, errT), vprCapacity)(pos, info, errT))(pos, info, errT)) // 0 <= cap
+            _ <- write(vpr.Exhale(vpr.LeCmp(vprLength, vprCapacity)(pos, info, errT))(pos, info, errT)) // len <= cap
+            _ <- errorT {
+              case err.ExhaleFailed(Source(info), _, _) => PreconditionError(info).dueTo(MakePreconditionError(info))
+            }
+
+            // inhale forall i: int :: {loc(a, i)} 0 <= i && i < [cap] ==> Footprint[ a[i] ]
+            footprintAssertion <- addressFootprint(ctx)(slice, in.FullPerm(slice.info))
+            _ <- write(vpr.Inhale(footprintAssertion)(pos, info, errT))
+
+            lenExpr = in.Length(slice)(makeStmt.info)
+            capExpr = in.Capacity(slice)(makeStmt.info)
+
+            // inhale cap(a) == [cap]
+            eqCap <- ctx.typeEncoding.equal(ctx)(capExpr, capArg, makeStmt)
+            _ <- write(vpr.Inhale(eqCap)(pos, info, errT))
+
+            // inhale len(a) == [len]
+            eqLen <- ctx.typeEncoding.equal(ctx)(lenExpr, lenArg, makeStmt)
+            _ <- write(vpr.Inhale(eqLen)(pos, info, errT))
+
+            // inhale forall i: int :: {loc(a, i)} 0 <= i && i < [len] ==> [ a[i] == dfltVal(T) ]
+            eqValueAssertion <- boundedQuant(
+              length = vprLength,
+              trigger = _ => Seq(), // TODO: add trigger
+              body = (x: in.BoundVar) =>
+                ctx.typeEncoding.equal(ctx)(in.IndexedExp(slice, x)(makeStmt.info), in.DfltVal(typeParam)(makeStmt.info), makeStmt)
+            )(makeStmt)(ctx)
+            _ <- write(vpr.Inhale(eqValueAssertion)(pos, info, errT))
+
+            ass <- ctx.typeEncoding.assignment(ctx)(in.Assignee.Var(target), slice, makeStmt)
+          } yield ass
+        )
+    }
+  }
+
+
+  /**
+    * Encodes the permissions for all addresses of a shared type,
+    * i.e. all permissions involved in converting the shared location to an exclusive r-value.
+    * An encoding for type T should be defined at all shared locations of type T.
+    *
+    * Footprint[loc: []T] -> forall idx :: {???} 0 <= idx < cap(l) ==> Footprint[ loc[idx] ]
+    *
+    * We do not use let because (at the moment) Viper does not accept quantified permissions with let expressions.
+    */
+  override def addressFootprint(ctx: Context): (in.Location, in.Permission) ==> CodeWriter[vpr.Exp] = {
+    case (loc :: ctx.Slice(t) / Shared, perm) =>
+      val (pos, info, errT) = loc.vprMeta
+
+      val fieldAccWriter = for {
+        vprPerm <- ctx.typeEncoding.expr(ctx)(perm)
+        l <- ctx.expr.translate(loc)(ctx).map(_.asInstanceOf[vpr.FieldAccess])
+      } yield vpr.FieldAccessPredicate(l, vprPerm)(pos, info, errT)
```

Comment:
This part should not be required for make. If you make `a` exclusive, then these permissions are not required.